File: hackerrank/practice/larrys-array.py

def solve(a):
	i = 1
	# count inversions in the array
	inv = 0
	f = {}
	f[a[-1]] = 0
	for i in xrange(len(a)-2,-1,-1):
		for j in xrange(i+1,len(a)):
			if a[j] == a[i]-1:
				f[a[i]] = f[a[j]] + 1
				inv += f[a[i]]
				print f
				break
			else:
				f[a[i]] = 0
	return not inv%2

for _ in xrange(int(raw_input())):
	n = raw_input()
	a = map(int,raw_input().split())
	print "YES" if solve(a) else "NO"
			

File: hackerrank/practice/larrys-array.cpp
#include <iostream>

using namespace std;

bool solve(int a[], int n){
	int inv = 0;
	for (int i = 0; i < n-1; i += 1){
		for (int j = i+1; j < n; j += 1){
			if (a[j]<a[i]){
				inv += 1;
			}
		}
	}
	if (inv%2 == 0) return true;
	return false;
}

int main(void){
	int t, n;
	cin >> t;
	while (t--){
		cin >> n;
		int a[n];
		for (int i = 0; i < n; i += 1){
			cin >> a[i] ;
		}
		bool ans = solve(a, n);
		cout << (ans ? "YES" : "NO" ) << endl;
	}
}